src/common/utils/asyncUtils.ts

```diff
@@ -1,3 +1,9 @@
 export async function sleep(milliseconds: number) {
     return new Promise<void>((resolve) => setTimeout(resolve, milliseconds));
 }
+
+export function timeout(ms: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+// TODO: Advanced timeout from core async: https://github.com/microsoft/vscode-python-environments/issues/953
```

Comment:
Looks like it's identical to `sleep`. I recommend renaming this to `timeout` so it's consistent with the core codebase since the core team works on this repo.